modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotCheckProcess.java

```diff
@@ -0,0 +1,740 @@
+            if (!F.isEmpty(nodeMetas)) {
+                onlineNodesConstIdsStr.add(node.consistentId().toString());
+
+                nodeMetas.forEach(nodeMeta -> metasPerRespondedNodes.computeIfAbsent(nodeMeta.consistentId(),
+                    m -> new TreeSet<>(Comparator.comparingLong(ClusterNode::order))).add(node));
+            }
+        });
+
+        String locNodeConsIdStr = locNode.consistentId().toString();
+        List<SnapshotMetadata> metasToProc = new ArrayList<>();
+
+        for (SnapshotMetadata meta : locMetas) {
+            if (meta.consistentId().equals(locNodeConsIdStr)) {
+                assert !metasToProc.contains(meta) : "Local snapshot metadata is already assigned to process";
+
+                metasToProc.add(meta);
+
+                continue;
+            }
+
+            if (!onlineNodesConstIdsStr.contains(meta.consistentId())
+                && F.first(metasPerRespondedNodes.get(meta.consistentId())).id().equals(kctx.localNodeId()))
+                metasToProc.add(meta);
+        }
+
+        return metasToProc;
+    }
+
+    /**
+     * Starts the snapshot validation process.
+     *
+     * @param snpName Snapshot name.
+     * @param snpPath Snapshot directory path.
+     * @param grpNames List of cache group names.
+     * @param fullCheck If {@code true}, additionally calculates partition hashes. Otherwise, checks only snapshot integrity
+     *                  and partition counters.
+     * @param incIdx Incremental snapshot index. If not positive, snapshot is not considered as incremental.
+     * @param allRestoreHandlers If {@code true}, all the registered {@link IgniteSnapshotManager#handlers()} of type
+     *                    {@link SnapshotHandlerType#RESTORE} are invoked. Otherwise, only snapshot metadatas and partition
+     *                    hashes are validated.
+     */
+    public IgniteInternalFuture<SnapshotPartitionsVerifyTaskResult> start(
+        String snpName,
+        @Nullable String snpPath,
+        @Nullable Collection<String> grpNames,
+        boolean fullCheck,
+        int incIdx,
+        boolean allRestoreHandlers
+    ) {
+        assert !F.isEmpty(snpName);
+
+        UUID reqId = UUID.randomUUID();
+
+        Set<UUID> requiredNodes = new HashSet<>(F.viewReadOnly(kctx.discovery().discoCache().aliveBaselineNodes(), node2id()));
+
+        SnapshotCheckProcessRequest req = new SnapshotCheckProcessRequest(
+            reqId,
+            requiredNodes,
+            snpName,
+            snpPath,
+            grpNames == null ? null : new HashSet<>(grpNames),
+            fullCheck,
+            incIdx,
+            allRestoreHandlers
+        );
+
+        GridFutureAdapter<SnapshotPartitionsVerifyTaskResult> clusterOpFut = new GridFutureAdapter<>();
+
+        clusterOpFut.listen(fut -> {
+            clusterOpFuts.remove(reqId);
+
+            if (log.isInfoEnabled())
+                log.info("Finished snapshot validation process [req=" + req + ']');
```

Comment:
In the other places we have `Finished snapshot validation [req=`. Let's unify and remove word 'process'.